### ISSUE TITLE
Slack target of receiver in AlertmanagerConfig it's not active

### ISSUE THREAD

**Issue by tbernacchi** [rancher/dashboard#7517]
**Setup**
- Rancher version: 2.6.9
- Browser type & version: Chrome Version 107.0.5304.110 (Official Build) (x86_64)

**Describe the bug**
When configuring a Slack target for an AlertmanangerConfig the fields 'Secret with Slack Webhook URL' and 'Key' it's not active.

**To Reproduce**
Following this procedure: https://www.suse.com/support/kb/doc/?id=000020737
...
...
...
-> Once added, we will need to Edit Config
-> From there, we can add a new Receiver (This is where we can specify which type of notifications to receive)
-> For Slack, select Add Slack
-> Under Secret with Slack Webhook URL, select the Secret name

**Result**
The fields 'Secret with Slack Webhook URL' and 'Key' it's not active. It's not possible to select the Secret name and the Key.

**Expected Result**
Be able to select the Secret name under 'Secret with Slack Webhook URL' field and 'Key'.

**Screenshots**
<img width="1347" alt="Screen Shot 2022-11-22 at 07 45 20" src="https://user-images.githubusercontent.com/26934246/203304981-98309490-2a41-449f-93fe-917b8027490e.png">

<img width="1347" alt="Screen Shot 2022-11-22 at 08 26 48" src="https://user-images.githubusercontent.com/26934246/203305024-009c68d7-47ab-4830-ba70-b9c3903b9ad8.png">

**Additional context**
The workaround it's to Click at top of the `AlertmanagerConfig` name, `Edit YAML`, and manually add the `key` and `name`:

```
spec:
  receivers:
    - name: test-github
      slackConfigs:
        - apiURL:
            key: test-tedyo
            name: test-tedyo
          channel: monitoring-slack-integrations

**Comment by richard-cox** [comment]
@tbernacchi The screenshots show the resource's view `Config` page which is read-only.

Can you confirm that the page shown after going to the resource menu's `Edit Config` has the correct editable fields?

**Comment by tbernacchi** [comment]
Hi @richard-cox - Yeap, I confirmed, and when I `Edit Config` and add the correct values from the dropdown and `Save` it's not persisting, only adding manually on 'Edit YAML'.

**Comment by richard-cox** [comment]
@tbernacchi @neuhausskaio To confirm, the problem is that changes to the resource via the form are not applied on save?

If so, are there any errors the the browser's console?

**Comment by neuhausskaio** [comment]
**To confirm, the problem is that changes to the resource via the form are not applied on save?**

Yes, when selecting the secret and the value key it's not saving the configuration, then I had to fill in the receiver information through the YAML file.

**Comment by tbernacchi** [comment]
[SURE-5622]

**Comment by nwmac** [comment]
The issue with disabled fields looks to have been addressed in 2.7.0.

There is an issue where there is no validation on the receiver name field and this is not shown as required - if you do not fill it in, the receiver will not be added.

**Comment by ronhorton** [comment]
**Pass** verified in rancher `v2.7.2-rc6`
1. installed alerting drivers
2. created alertmanager config
3. added slack config receiver
4. target (secret with slack URL) is active & saves value
